src/v2/stack_aggregator.py

```diff
@@ -246,22 +247,130 @@ def _get_package_details_with_vulnerabilities(self) -> List[Dict[str, object]]:
+
+                    # Get number of vulnerabilities ignored in current package.
+                    ignored_vulnerability_count = vuln_before_ignoring - vuln_after_ignoring
             else:
                 continue  # pragma: no cover
             transitive_details = []
+
             for transitive in transitives:
                 if transitive in self._normalized_packages.dependency_graph:
                     continue
                 transitive_detail = self._normalized_package_details.get(transitive)
+
                 if _has_vulnerability(transitive_detail):
                     transitive_detail = transitive_detail.copy()
+
+                    # Number of transitive vulnerabilities before ignoring.
+                    trans_vuln_before_ignoring = \
+                        len(transitive_detail.public_vulnerabilities) + \
+                        len(transitive_detail.private_vulnerabilities)
+
+                    # User asked to ignore all vulnerabilities of direct dependency
+                    # hence all transitive vulnerabilities are also ignored.
+                    if ignore_all_transitives:
+                        ignored_trans_vulnerability_count = trans_vuln_before_ignoring
+                        continue
+
+                    # Ignore selected vulnerabilities
+                    trans_public_vulnerabilities = [
+                        vuln for vuln in transitive_detail.public_vulnerabilities
+                        if vuln.id not in all_vulnerabilities_to_ignore]
+                    trans_private_vulnerabilities = [
+                        vuln for vuln in transitive_detail.private_vulnerabilities
+                        if vuln.id not in all_vulnerabilities_to_ignore]
+
+                    trans_public_vuln_after_ignoring = 0
+                    if trans_public_vulnerabilities:
+                        # Reassign new public vulnerabilities after ignoring selected IDs.
+                        transitive_detail.public_vulnerabilities = trans_public_vulnerabilities
+
+                        # Number of public vulnerabilities after ignoring.
+                        trans_public_vuln_after_ignoring = \
+                            len(transitive_detail.public_vulnerabilities)
+
+                    trans_private_vuln_after_ignoring = 0
+                    if trans_private_vulnerabilities:
+                        # Reassign new private vulnerabilities after ignoring selected IDs.
+                        transitive_detail.private_vulnerabilities = trans_private_vulnerabilities
+
+                        # Number of private vulnerabilities after ignoring.
+                        trans_private_vuln_after_ignoring = \
+                            len(transitive_detail.private_vulnerabilities)
+
+                    # Get number of transitive vulnerabilities ignored in current package.
+                    ignored_trans_vulnerability_count = trans_vuln_before_ignoring - (
```

Comment:
I think ignored transitive vulnerabilities should be added to `ignored_trans_vulnerability_count` instead of it being overwritten here too.